src/lib/eth.ts

```diff
@@ -83,10 +103,10 @@ export function isValidChainId(chainId: string | number) {
 }
 
 export function getChainIdByNetwork(network: Network) {
-  const connectedChainId = getConnectedProviderChainId()
-  if (!connectedChainId) {
-    throw new Error('Could not get connected provider chain id')
+  const appChainId = _getAppChainId()
```

Comment:
This is the fix to the issue with `getChainIdByNetwork` which causes, amog other things, the `NetworkCheck` component to fail to detect if the network is wrong.